.github/workflows/link-check-deploy.yml

```diff
@@ -2,47 +2,58 @@ name: Check new links against deployment
 # This workflow "triggers" and skips on deployment because GitHub Actions /
 # Checks refuses to show the check on deployment_status
 on:
-  - deployment
   - deployment_status
 jobs:
   run:
-    name: Initialize
+    name: Run
     runs-on: ubuntu-latest
+    continue-on-error: true
     if:
       github.event.deployment.ref != 'main' &&
       github.event.deployment_status.state == 'success'
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           fetch-depth: 0
-      - id: build_check
-        uses: LouisBrunner/checks-action@v1.0.0
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          name: Report
-          status: queued
       - name: Run Link Check
         id: check
-        uses: iterative/link-check.action@v0.11
+        run: |
+          set +e
+          body="<h1 id=\"link-check\">Link Check Report</h1>
+
+          $(
+            npx repo-link-check \
+            -d -c config/link-check/config.yml \
+            -r ${{ github.event.deployment.payload.web_url }}
+          )"
+          body="${body//'%'/'%25'}"
+          body="${body//$'\n'/'%0A'}"
+          body="${body//$'\r'/'%0D'}"
+          echo "::set-output name=report::$body"
+          exit 0
+
+      - name: Find PR associated with deployment
+        uses: actions/github-script@v6
+        id: pr-number
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
-          diff: main
-          configFile: config/link-check/config.yml
-          rootURL: '${{ github.event.deployment.payload.web_url }}'
-          output: checksAction
-      - uses: LouisBrunner/checks-action@v1.0.0
-        if: ${{ success() }}
+          script: |
+            const findPR = require('./.github/workflows/deployment-report.js')
+            return await findPR({ github, context, core })
```

Comment:
This action is really handy for rapidly prototyping weird stuff with the GitHub API. You can import modules so most of the code can be properly linted instead of being stuck in a YAML block.